src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,133 @@
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.SerialPort.Port;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import com.kauailabs.navx.frc.AHRS;
+
+public class Drivetrain extends SubsystemBase {
+  private WPI_TalonFX driveLeftMotor1,driveLeftMotor2,driveRightMotor1,driveRightMotor2;
+  private AHRS gyro;
+
+  public Drivetrain() {
+
+    //creating motors
+    driveLeftMotor1 = new WPI_TalonFX(Constants.driveTalonLeftMotor1);
+    driveLeftMotor2 = new WPI_TalonFX(Constants.driveTalonLeftMotor2);
+    driveRightMotor1 = new WPI_TalonFX(Constants.driveTalonRightMotor1);
+    driveRightMotor2 = new WPI_TalonFX(Constants.driveTalonRightMotor2);
+
+    //making right motors go right
+    driveRightMotor1.setInverted(true);
+    driveRightMotor2.setInverted(true);
+    driveLeftMotor1.setInverted(false);
+    driveLeftMotor2.setInverted(false);
+
+    gyro = new AHRS(Port.kUSB);
+
+    resetGyro();
+  }
+
+
+  //gyro tingz
+  public void resetGyro(){
+    gyro.reset();
+  }
+  public double getGyroAngle(){
+    return(getRawGyroAngle() % 360 + 360) % 360;
+  }
+  public double getRawGyroAngle() {
+    return gyro.getAngle();
+  }
+
+  //creating percent output for both right and left
+  public void drivePercentOutput(double speed){
+    drivePercentOutput(speed, speed);
+  }
+  public void drivePercentOutput(double leftSpeed, double rightSpeed){
+    driveLeftPercentOutput(leftSpeed);
+    driveRightPercentOutput(rightSpeed);
+  }
+  public void driveLeftPercentOutput(double speed){
+    driveLeftMotor1.set(ControlMode.PercentOutput, speed);
+  }
+  public void driveRightPercentOutput(double speed){
+    driveRightMotor1.set(ControlMode.PercentOutput, speed);
+  }
+
+  //creating drive velocity for both right and left
+  public void driveVelocity(double speed){
+    driveVelocity(speed, speed);
+  }
+  public void driveVelocity(double leftSpeed, double rightSpeed){
+    driveLeftVelocity(leftSpeed);
+    driveRightVelocity(rightSpeed);
+  }
+  public void driveLeftVelocity(double speed){
+    driveLeftMotor1.set(ControlMode.Velocity, speed);
+  }
+  public void driveRightVelocity(double speed){
+    driveRightMotor1.set(ControlMode.Velocity, speed);
+  }
+
+  //creative drive voltage for both right and left
+  public void driveVoltage(double outputVolts){
+    driveVoltage(outputVolts, outputVolts);
+  }
+  public void driveVoltage(double leftOutputVolts, double rightOutputVolts){
+    driveRightVoltage(rightOutputVolts);
+    driveLeftVoltage(leftOutputVolts);
+  }
+  public void driveRightVoltage(double outputVolts){
+    driveRightMotor1.setVoltage(outputVolts);
+  }
+  public void driveLeftVoltage(double outputVolts){
+    driveLeftMotor1.setVoltage(outputVolts);
+  }
+
+  //robot can stop
+  public void stop(){
+    driveVelocity(0, 0);
```

Comment:
Change to drivePercentOutput(0,0). Velocity mode had a loop and therefore will have a hard time with an untuned loop.